vector/src/main/java/im/vector/app/core/extensions/Context.kt

```diff
@@ -77,3 +83,31 @@ val Context.dataStoreProvider: (String) -> DataStore<Preferences> by dataStorePr
 fun Context.safeOpenOutputStream(uri: Uri): OutputStream? {
     return contentResolver.openOutputStream(uri, "wt")
 }
+
+/**
+ * Checks for an active connection to infer if the device is offline.
+ * This is useful for breaking down UnknownHost exceptions and should not be used to determine if a valid connection is present
+ *
+ * @return true if no active connection is found
+ */
+@Suppress("deprecation")
+@SuppressLint("NewApi") // false positive
+fun Context.inferNoConnectivity(buildMeta: BuildMeta): Boolean {
+    val connectivityManager = getSystemService<ConnectivityManager>()!!
+    return if (buildMeta.sdkInt > Build.VERSION_CODES.M) {
+        val networkCapabilities = connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+        when {
+            networkCapabilities?.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) == true -> false
+            networkCapabilities?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true     -> false
+            networkCapabilities?.hasTransport(NetworkCapabilities.TRANSPORT_VPN) == true      -> false
+            else                                                                              -> true
```

Comment:
This might be a silly question/use case but should we check the internet connection over `TRANSPORT_ETHERNET` and `TRANSPORT_BLUETOOTH`?

Reply:
good catch! I will double check if `hasTransport` considers bluetooth as an internet provider or simply if a bluetooth connection is present